src/plugins/omemo/Plugin.ts

```diff
@@ -100,6 +103,42 @@ export default class OMEMOPlugin extends EncryptionPlugin {
       });
    }
 
+   private getDeviceListFromContact(contact: IContact)
+   {
+      let connection = this.pluginAPI.getConnection();
+      let _self = this;
+      return connection.getPEPService().retrieveItems('eu.siacs.conversations.axolotl.devicelist',contact.getJid().bare).then(function(data) {
+
+            return new Promise(function(resolve, reject) {
+                if (_self.onDeviceListUpdate(data))
+                {
+                    const promises = [];
+                    let arr = $(data).find("device");
+                    let i=0;
+                    for (; i < arr.length; i++) {
+                       promises.push(connection.getPEPService().retrieveItems('eu.siacs.conversations.axolotl.bundles:'+$(arr[i]).attr('id'),contact.getJid().bare).then(function(data) {
```

Comment:
No need to query all bundles. If a contact has a device list this should be enough.

Reply:
I tried withouth receiving the bundles only the key, then i got this error that my contact does not support omemo.
Then i notice in the omemo device/key dialog that the keys are not loaded.
If i force to load the keys, then they are avaiable...